### PR TITLE
[MSHARED-1035] Get rid of maven-plugin-testing-tools for IT test

### PR DESCRIPTION
**Pull request by slawekjaranowski**
IT tests will be executed by maven-invoker-plugin

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes. Also be sure having selected the correct component.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by michael-o** [comment]
That's massive..., will run...

File: pom.xml [hunk 1]
```diff
@@ -192,24 +149,34 @@
...
+              <cloneProjectsTo>${project.build.directory}/it</cloneProjectsTo>
+              <localRepositoryPath>target/local-repo</localRepositoryPath>
+              <settingsFile>src/it/settings.xml</settingsFile>
+              <postBuildHookScript>verify</postBuildHookScript>
+              <goals>
+                <goal>verify</goal>
+              </goals>
+              <properties>
+                <maven.compiler.source>${java.specification.version}</maven.compiler.source>
+                <maven.compiler.target>${java.specification.version}</maven.compiler.target>
```
Comment (michael-o) [inline pom.xml:177]: Does this really make sense? I mean it depends now on the Java version run?!

**Comment by slawekjaranowski** [comment]
I can't use new version of `maven-plugin-testing-harness` ... because for tests MavenProject instance is needed ... 
MavenProject must contains resolve all dependency which is difficult to mock

File: pom.xml (hunk 1 above)
Comment (slawekjaranowski) [reply pom.xml:177]: Many thing depends on bytecode version in analysis so compiling test code for many version can help detect future problem with ... next java version

if we need test feature which depends of jdk version we can add such requrement in invoker.properties

Comment (michael-o) [reply pom.xml:177]: Makes sense.

**Review by michael-o — APPROVED** [review]
Tests pass!

**Comment by slawekjaranowski** [comment]
All test pass ... on GH and Jenkins
even with 18-ea on localhost 😸

**Review by slachiewicz — APPROVED** [review]

**Review by elharo — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #204

**Pull request merged**